rllib/core/models/torch/primitives.py

```diff
@@ -165,18 +191,18 @@ def __init__(
             in_size = out_size
             in_depth = out_depth
 
-        self.output_width, self.output_height = out_size
-        self.output_depth = out_depth
-
         # Create the CNN.
         self.cnn = nn.Sequential(*layers)
 
         self.expected_input_dtype = torch.float32
 
     def forward(self, inputs):
-        # Permute b/c data comes in as [B, dim, dim, channels]:
+        # Permute b/c data comes in as channels_last ([B, dim, dim, channels]) ->
+        # Convert to `channels_first` for torch:
         inputs = inputs.permute(0, 3, 1, 2)
-        return self.cnn(inputs.type(self.expected_input_dtype))
+        out = self.cnn(inputs.type(self.expected_input_dtype))
+        # Permute back to `channels_last`.
+        return out.permute(0, 2, 3, 1)
```

Comment:
Damn 😄

Reply:
It was fine b/c we had the flatten+dense anyways, but now that the dense is gone and flatten is optional, I wanted to have a unified behavior on the surface.